Userland/Libraries/LibWeb/Page/Page.cpp

```diff
@@ -290,6 +291,28 @@ void Page::accept_dialog()
     }
 }
 
+void Page::did_request_color_picker(WeakPtr<HTML::HTMLInputElement> target)
+{
+    if (m_pending_non_blocking_dialog == PendingNonBlockingDialog::None) {
+        m_pending_non_blocking_dialog = PendingNonBlockingDialog::ColorPicker;
+        m_pending_non_blocking_dialog_target = move(target);
+
+        m_client.page_did_request_color_picker();
+    }
+}
+
+void Page::color_picker_closed(Optional<String> response)
+{
+    if (m_pending_non_blocking_dialog == PendingNonBlockingDialog::ColorPicker) {
+        m_pending_non_blocking_dialog = PendingNonBlockingDialog::None;
+
+        if (m_pending_non_blocking_dialog_target.has_value() && !m_pending_non_blocking_dialog_target.value().is_null()) {
+            m_pending_non_blocking_dialog_target.value()->did_pick_color(move(response));
```

Comment:
To safely access a `WeakPtr`'s value, use something like:

```c++
if (auto ref = weak_ptr.strong_ref())
    do_something_with(ref);
```

That guarantees that if the object isn't null, it'll stay in existence while you're using it.

Reply:
As per [this Discord discussion](https://discord.com/channels/830522505605283862/830739873119207426/1147953305713123439), "HTML element classes are not reference counted (they are garbage collected), so you can't call strong_ref on WeakPtr<Element>. That will give you RefPtr<Element>, which won't compile as you've seen.". By the event-loop architecture I assumed that any code to delete said element is not gonna run at the same time as the function which accesses it.

Reply:
:laughing: Wow. So I remembered someone talking about it, but didn't remember they were saying NOT to do what I suggested!

I'm not familiar with the GC system so I'll leave it to someone else to judge if this is ok. But it probably is. :^)

Reply:
@trflynn89 Can you confirm this usage is correct?

Reply:
Sounds good thanks